org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/AutoEditsPreferencePage.scala

```diff
@@ -30,16 +31,29 @@ import org.scalaide.util.eclipse.SWTUtils._
 /** This class is referenced through plugin.xml */
 class AutoEditsPreferencePage extends PreferencePage with IWorkbenchPreferencePage {
 
-  private val prefStore = IScalaPlugin().getPreferenceStore()
-
   private var descriptionArea: Text = _
   private var configurationArea: Text = _
   private var viewer: CheckboxTableViewer = _
 
   private val settings = AutoEdits.autoEditSettings.toArray
 
-  private var changes = Set[AutoEditSetting]()
-  private var configurations = Set[(AutoEditSetting, String)]()
+  private val prefStore = {
+    val ps = IScalaPlugin().getPreferenceStore
+    val keys = settings flatMap { s ⇒
+      import OverlayPreferenceStore._
+      Seq(
+        new OverlayKey(BOOLEAN, s.id),
+        new OverlayKey(STRING, s.configId)
+      )
+    }
+    val store = new OverlayPreferenceStore(ps, keys.toArray)
+    settings foreach { s ⇒
```

Comment:
You could use `store.load()`. Have a look at `FormatterPreferencePage`, it's using the same technique